pkg/lib/cockpit.js

```diff
@@ -1483,17 +1483,27 @@ function factory() {
             });
     };
 
-    function format_units(number, suffixes, factor, options) {
-        // backwards compat: "options" argument position used to be a boolean flag "separate"
-        if (!is_object(options))
-            options = { separate: options };
+    function format_units(suffixes, number, second_arg, third_arg) {
+        let options = second_arg;
 
+        /* compat API: we used to accept 'factor' as a separate second arg */
+        if (third_arg || (second_arg && !is_object(second_arg))) {
+            options = third_arg;
+
+            // double backwards compat: "options" argument position used to be a boolean flag "separate"
+            if (!is_object(options))
+                options = { separate: options };
+
+            options.factor = second_arg;
+        }
+
+        const factor = options?.factor || (options?.base2 ? 1024 : 1000);
```

Comment:
We never supported anything else than 1000 or 1024 here, and I have a rather strong feeling about *not* adding support for a general `{ factor: n}` option.

Reply:
The issue is that `factor` here could be a string like `"MB"` and we need to keep supporting that.  cockpit-podman is still using that...

Reply:
Aah, that, right! (Gosh, this is too complex/magic!)